src/utils/InputHandler.js

```diff
@@ -0,0 +1,14 @@
+import {Console} from '@woowacourse/mission-utils';
+import ErrorCode from "../datas/ErrorCode.js";
+
+class InputHandler {
+    static async getInput(instruction, processInput = (x) => x) {
+        const input = await Console.readLineAsync(instruction)
+        if (input.length <= 0)
+            throw new Error(ErrorCode.EMPTY_INPUT)
+        return processInput(input)
+    }
```

Comment:
```suggestion
    static async getInput(instruction, processInput) {
      const input = await Console.readLineAsync(instruction)
      if (input.length <= 0){
        throw new Error(ErrorCode.EMPTY_INPUT);
      }
          
      if(processInput){ // 포맷 함수가 있다면 실행 후 반환
        return processInput(input);
      }
          
      return input;
    }
```
인풋을 입력하고 추후에 포맷하는 로직을 고차 함수로 받는게 인상적이었던 것 같네요!

하지만 `processInput`에 Default Parameter를 사용해서 빈 함수를 만드는 것보다 `processInput` 있다면 함수를 실행하고 없다면 바로 반환하는게 조금 더 목적에 맞을 것 같다고 생각하는데 유정님 생각은 어떠신가요??

Reply:
무의미하게 실행되는 것보다 훨씬 좋은 방식인것 같아요! 
감사합니다 ㅎㅎ